features/setting_constraints/matching_arguments.feature

```diff
@@ -72,6 +72,36 @@ Feature: Matching arguments
       | expected: (a collection containing exactly 1 and 2)           |
       | got: ([1, 3])                                                 |
 
+  Scenario: Using satisfy for complex custom expecations
+    Given a file named "rspec_satisfy_spec.rb" with:
+      """ruby
+      RSpec.describe "Using satisfy for complex custom expecations" do
+        let(:dbl) { double }
+        before do
+          expectation = satisfy do |data|
+            data[:a][:b][:c] == 5
+          end
+          expect(dbl).to receive(:foo).with(expectation)
+        end
+
+        it "passes when the expectation is true" do
+          dbl.foo(a: { b: { c: 5 } })
+        end
+
+        it "fails when the expectation is false" do
+          dbl.foo(a: { b: { c: 3 } })
+        end
```

Comment:
It is very subjective, but I would argue about this example's style. Firstly, the expectation is made in a hook.
The term `expectation` from my perspective is used incorrectly, the `satisfy` matcher is used as an argument matcher here.

How do you feel about this instead:
```ruby
      RSpec.describe "Using satisfy for complex custom expecations" do
        let(:dbl) { double }

        def a_b_c_equals_5
          satisfy { |data| data[:a][:b][:c] == 5 }
        end

        it "passes when the expectation is true" do
          expect(dbl).to receive(:foo).with(a_b_c_equals_5)
          dbl.foo(a: { b: { c: 5 } })
        end

        it "fails when the expectation is false" do
          expect(dbl).to receive(:foo).with(a_b_c_equals_5)
          dbl.foo(a: { b: { c: 3 } })
        end
```

Reply:
> It is very subjective, but I would argue about this example's style. Firstly, the expectation is made in a hook. 
> ...
> How do you feel about this instead:

I followed the style of other stories in this file, but I like yours better, so I adopted your suggestion.


> The term `expectation` from my perspective is used incorrectly, the `satisfy` matcher is used as an argument matcher here.
 
As I understand it, the goal is to verify an expectation. The mechanism is to use a matcher. If I've mixed things up, or if you want to emphasise one or the other in the documentation, let me know what changes you'd like to see.